C_routing.md

```diff
@@ -826,62 +799,36 @@ defmodule HelloPhoenix.Router do
-
-Here's how we might customize the `:before` pipeline. The example assumes that we have already defined the plugs `:authenticate` and `:set_current_user`.
-
-```elixir
-defmodule HelloPhoenix.Router do
-  use Phoenix.Router
-
-  pipeline :before do
-    plug :super
-    plug :authenticate
-    plug :set_current_user
-  end
-
-  pipeline :browser do
-    plug :accepts, ~w(html)
-    plug :fetch_session
-    plug :protect_from_forgery
-  end
-
-  pipe_through :browser
-
-  resources "reviews", HelloPhoenix.ReviewController
-end
-```
-
-`plug :super` will invoke any plugs in the `:browser` pipeline before it plugs `:authenticate` and `:set_current_user`.
-
 ###Channel Routes
 
 Channels are a very exciting, realtime component of the Phoenix framework. They are so important that they will have a guide of their own.
 
 Channels are roughly analogous to controllers except that they are capable of bi-directional communication and their connections persist beyond the initial response. They are also closely tied to a client - written for JavaScript, iOS or Android. For now, we'll focus on defining routes for them and leave a detailed discussion of their capabilities to the Channel Guide.
 
-Each channel depends on a socket mounted at a given point for its communication. The first thing we need to do to define a route for a channel. Then we define a socket and specify the path to its mount point.
+Each channel depends on a socket mounted at a given point for its communication. We can define the socket in a way that looks a lot like a scope for a regular route. Which is to say we define a socket block with a path to the socket's mount point and the name of our application to fully qualify our channel name.
 
 Here's what that looks like in our router file.
 
 ```elixir
 defmodule HelloPhoenix.Router do
   use Phoenix.Router
-  use Phoenix.Router.Socket, mount: "/my_socket"
+
+  socket "/ws", HelloPhoenix do
+  end
+  ...
 end
 ```
 
-The next thing we need to do is define a channel, giving it a name and associating it with a channel module which will implement its behavior. If we have a channel module called "OurChannel" and a channel called `our_channel_name`, the code to do this is straightforward, `channel "our_channel_name", HelloPhoenix.OurChannel`
-
-The whole router, then, looks like this.
+Next, we need to define a channel, specifying a topic and associating it with the channel module which will implement its behavior. If we have a channel module called `RoomChannel` and a topic called `lobby`, the code to do this is straightforward.
 
 ```elixir
 defmodule HelloPhoenix.Router do
   use Phoenix.Router
-  use Phoenix.Router.Socket, mount: "/my_socket"
 
-  channel "our_channel_name", HelloPhoenix.OurChannel
+  socket "/ws", HelloPhoenix do
+    channel "lobby", RoomChannel
```

Comment:
No, they can just be normal strings, but the idiom is "topic:subtopic", so a wildcard pattern is the preferred approach since it pretty much mirrors what we were doing prior to 0.8, but we were conflating what a channel and a topic was. Now, the topic is just a string identifier, and we dispatch to channels based on the topic patterns. The new channel docs should help describe the changes https://github.com/phoenixframework/phoenix/blob/master/lib/phoenix/channel.ex

"bare" topics make sense for some use-cases, like a "system" topic where you have just one singleton thing, i.e.:

```elixir
channel "system", SystemChannel
...
def join("system", message, socket) ...

def handle_in("set:clock", %{"time" => time})...
```

Maybe. Wildcards offer much more flexibility and is where I'll steer folks towards unless there's a good reason not to.